samples/managed-app-sample/mainTemplate.json

```diff
@@ -0,0 +1,282 @@
+            "metadata": {
+                "description": "VM count"
+            }
+        },
+        "dnsPrefix": {
+            "defaultValue": "[concat(parameters('clusterResourceName'),'-dns')]",
+            "type": "String",
+            "metadata": {
+                "description": "Optional DNS prefix to use with hosted Kubernetes API server FQDN."
+            }
+        },
+        "osDiskSizeGB": {
+            "defaultValue": 0,
+            "minValue": 0,
+            "maxValue": 1023,
+            "type": "Int",
+            "metadata": {
+                "description": "Disk size (in GiB) to provision for each of the agent pool nodes. This value ranges from 0 to 1023. Specifying 0 will apply the default disk size for that agentVMSize."
+            }
+        },
+        "kubernetesVersion": {
+            "type": "String",
+            "defaultValue": "1.24.0",
+            "metadata": {
+                "description": "The version of Kubernetes."
+            }
+        },
+        "networkPlugin": {
+            "defaultValue": "kubenet",
+            "allowedValues": [
+                "azure",
+                "kubenet"
+            ],
+            "type": "String",
+            "metadata": {
+                "description": "Network plugin used for building Kubernetes network."
+            }
+        },
+        "enableRBAC": {
+            "defaultValue": true,
+            "type": "Bool",
+            "metadata": {
+                "description": "Boolean flag to turn on and off of RBAC."
+            }
+        },
+        "enablePrivateCluster": {
+            "defaultValue": false,
+            "type": "Bool",
+            "metadata": {
+                "description": "Enable private network access to the Kubernetes cluster."
+            }
+        },
+        "enableHttpApplicationRouting": {
+            "defaultValue": true,
+            "type": "Bool",
+            "metadata": {
+                "description": "Boolean flag to turn on and off http application routing."
+            }
+        },
+        "enableAzurePolicy": {
+            "defaultValue": false,
+            "type": "Bool",
+            "metadata": {
+                "description": "Boolean flag to turn on and off Azure Policy addon."
+            }
+        },
+        "enableSecretStoreCSIDriver": {
+            "defaultValue": false,
+            "type": "Bool",
+            "metadata": {
+                "description": "Boolean flag to turn on and off secret store CSI driver."
+            }
+        }
+    },
+    "variables": {
+        "plan-name": "REPLACE-WITH-YOUR-PLAN-ID",
+        "plan-publisher": "REPLACE-WITH-YOUR-PUBLISHER-ID",
+        "plan-offerID": "REPLACE-WITH-YOUR-OFFER-ID",
+        "releaseTrain": "REPLACE-WITH-YOUR-RELEASE-TRAIN",
+        "clusterExtensionTypeName": "REPLACE-WITH-YOUR-EXTENSION-TYPE-NAME"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.ManagedIdentity/userAssignedIdentities",
```

Comment:
curious, would CPA tool allow adding a managed identity ? I remember we only allow AKS cluster and extension resources creation

Reply:
no it doesn't. This mainTemplate and UiDefinition is for managed app.